src/core/service.c

```diff
@@ -5003,6 +5062,137 @@ static int service_can_clean(Unit *u, ExecCleanMask *ret) {
+                                        unit_pid,
+                                        propagate_directory,
+                                        "/run/systemd/incoming/",
+                                        src, dst,
+                                        read_only,
+                                        make_file_or_directory,
+                                        options,
+                                        c->mount_image_policy ?: &image_policy_service);
+                else
+                        r = bind_mount_in_namespace(
+                                        unit_pid,
+                                        propagate_directory,
+                                        "/run/systemd/incoming/",
+                                        src, dst,
+                                        read_only,
+                                        make_file_or_directory);
+                if (r < 0)
+                        log_unit_warning_errno(
+                                        u,
+                                        r,
+                                        "Failed to mount '%s' on '%s' in unit's namespace: %m",
+                                        src,
+                                        dst);
+                else
+                        log_unit_debug(u, "Mounted '%s' on '%s' in unit's namespace", src, dst);
+                exit(r < 0 ? EXIT_FAILURE : EXIT_SUCCESS);
+        }
+
+        r = unit_watch_pidref(u, &worker, /* exclusive= */ true);
+        if (r < 0)
+                return r;
+
+        sd_bus_message_unref(s->mount_request); /* Just in case */
+        s->mount_request = sd_bus_message_ref(message);
+        s->control_pid = TAKE_PIDREF(worker);
+        service_set_state(s, SERVICE_MOUNTING);
+        return 0;
+
+fail:
+        s->mount_result = SERVICE_FAILURE_RESOURCES;
+        s->timer_event_source = sd_event_source_disable_unref(s->timer_event_source);
+        return r;
+}
+
+static bool service_can_live_mount(Unit *u, const char *dst) {
+        assert(u);
+        assert(dst);
+
+        /* If it would be dropped at startup time, return an error. The context should always be available, but
+         * there's an assert in exec_needs_mount_namespace, so double-check just in case. */
+        ExecContext *c = unit_get_exec_context(u);
+        if (!c) {
+                log_unit_debug(u, "Cannot access unit execution context");
+                return false;
+        }
+        if (path_startswith_strv(dst, c->inaccessible_paths)) {
+                log_unit_debug(u, "%s is not accessible to this unit", dst);
+                return false;
+        }
+
+        /* Ensure that the unit was started in a private mount namespace */
+        if (!exec_needs_mount_namespace(c, NULL, unit_get_exec_runtime(u))) {
+                log_unit_debug(u, "Unit not running in private mount namespace, cannot activate bind mount");
+                return false;
```

Comment:
Hmm? Why is the restriction for credentials removed?

Reply:
rebase gone wrong